src/layouts/Login/SgidLoginCallbackPage.tsx

```diff
@@ -1,14 +1,127 @@
+import {
+  Flex,
+  Text,
+  Icon,
+  VStack,
+  Spinner,
+  Box,
+  Divider,
+} from "@chakra-ui/react"
+import { Link } from "@opengovsg/design-system-react"
+import { useEffect } from "react"
+import { BiChevronRight } from "react-icons/bi"
 import { useLocation } from "react-router-dom"
 
-import { useSgidLogin } from "hooks/loginHooks"
+import { useSgidLogin, useSgidMultiuserLogin } from "hooks/loginHooks"
+
+import { IsomerLogo } from "assets"
 
 export const SgidLoginCallbackPage = (): JSX.Element => {
   const { search } = useLocation()
   const urlSearchParams = new URLSearchParams(search)
   const params = Object.fromEntries(urlSearchParams.entries())
   const { code } = params
 
-  useSgidLogin({ code })
+  const { data: employments } = useSgidLogin({ code })
+
+  const { mutateAsync: onSelectLogin } = useSgidMultiuserLogin()
+
+  useEffect(() => {
+    if (!employments) return
+    if (employments.length === 1) {
+      window.location.replace("/sites")
```

Comment:
again, `useHistory`

Reply:
i assume the length of 1 is because they've already successfully validated and don't need to go through the multi user login flow?